src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -120,6 +120,9 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
     configure_logging: NotRequired[bool]
     """If True, the crawler will set up logging infrastructure automatically."""
 
+    max_crawl_depth: NotRequired[int | None]
+    """Maximum crawl depth. If set, the crawler will stop crawling after reaching this depth."""
```

Comment:
We should probably elaborate on the edge cases - for example, with `max_crawl_depth = 3`, do we process three or four "levels" of links? I'd assume that the start requests have `crawl_depth = 0`, and then we go all the way up to 3 and don't enqueue any further links, but it would be much better to have that stated explicitly in the docs.

Reply:
Yes, what you propose is perfect.